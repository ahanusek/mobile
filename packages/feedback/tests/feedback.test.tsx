import * as React from 'react';
import { render, mount } from 'enzyme';
import { html } from 'cheerio';
import TouchFeedback from '../src';

describe('basic', () => {
  it('base.', () => {
    const wrapper = render(
      <TouchFeedback
        activeClassName="active"
        activeStyle={{ color: 'red' }}
        disabled
      >
        <div style={{ color: '#000' }} className="normal">
          click to active
        </div>
      </TouchFeedback>,
    );
    expect(html(wrapper)).toMatchSnapshot();
  });

  it('works ok', async () => {
    const instance = mount(
      <TouchFeedback activeClassName="active" activeStyle={{ color: 'red' }}>
        <div style={{ color: '#000' }} className="normal">
          click to active
        </div>
      </TouchFeedback>,
    );

    const d = instance.find(TouchFeedback);
    const n = d.find('div').getDOMNode();

    d.simulate('touchstart');
    expect(d.state()).toHaveProperty('active', true);
    expect(n.className).toBe('normal active');
    expect(n.getAttribute('style')).toBe('color: red;');

    d.simulate('touchend');
    expect(d.state()).toHaveProperty('active', false);
    expect(n.className).toBe('normal');
    expect(n.getAttribute('style')).toBe('color: rgb(0, 0, 0);');
  });

  it('activeStyle false', async () => {
    const instance = mount(
      <TouchFeedback activeClassName="active" activeStyle={false}>
        <div style={{ color: '#000' }} className="normal">
          click to active
        </div>
      </TouchFeedback>,
    );

    const d = instance.find(TouchFeedback);
    const n = d.find('div').getDOMNode();

    d.simulate('touchstart');
    expect(n.className).toBe('normal');
  });

  it('disabled', async () => {
    const instance = mount(
      <TouchFeedback
        activeClassName="active"
        activeStyle={{ color: 'red' }}
        disabled
      >
        <div style={{ color: '#000' }} className="normal">
          click to active
        </div>
      </TouchFeedback>,
    );

    const d = instance.find(TouchFeedback);
    const n = d.find('div').getDOMNode();

    d.simulate('touchstart');
    expect(n.className).toBe('normal');
    expect(n.getAttribute('style')).toBe('color: rgb(0, 0, 0);');
  });
});
