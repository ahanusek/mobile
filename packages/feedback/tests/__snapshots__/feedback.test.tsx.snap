// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic base. 1`] = `"<div style=\\"color:#000\\" class=\\"normal\\">click to active</div>"`;
